### PR TITLE
Update documentation for Kubernetes 1.26+ and PCaaS2

### PR DESCRIPTION
**Pull request by huxcrux**
Fixes: https://github.com/elastx/issue-tracker/issues/356
Fixes: https://github.com/elastx/cluster-api/issues/148

### CODE REVIEW

File: content/en/docs/private-kubernetes/guides/loadbalancer.md [hunk 1]
```diff
@@ -207,3 +207,7 @@ zones. This is a limitation imposed by the OpenStack Octavia project.
 Reconfiguring the load balancers using annotations is not as dynamic and smooth
 as one would hope. For now, to change the configuration of a load balancer the
 service needs to be deleted and a new one created.
+
+## Loadbalancer protocols
+
+Loadbalancers are by default proxying TCP traffic however can be set to proxy UDP traffic, haproxys proxy protocal and even set to HTTP and HTTPS mode meaning the loadbalancer will act as a reverse proxy. We strongly recommend to use loadbalancers in TCP, UDP or haproxys proxy-protocol mode. This since http and https modes have quite a few limitations and does not perfom as well. Instead of using http and https mode in your loadbalancer we recommend point your loadbalance towards an ingress controller and let the ingress controller manage all proxying and TLS.
```
Comment (holmsten) [inline content/en/docs/private-kubernetes/guides/loadbalancer.md:213]: This section is a bit hard to follow and contain some spelling errors.

**Review by holmsten — CHANGES_REQUESTED** [review]
Found some bits that should be fixed but in general it looks good!

**Review by holmsten — APPROVED** [review]

File: content/en/docs/private-kubernetes/changelog/k8s-1.26.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+title: "Changelog for Kubernetes 1.26"
+description: "Changelog for Kubernetes 1.26"
+alwaysopen: true
+---
+
+## Versions
+
+The deployed Kubernetes version varies based on when your cluster is deployed. We try deploying cluster using the lasted patch release of Kubernetes.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/changelog/k8s-1.26.md:9]: Should probably be "latest" instead of "lasted"

File: content/en/docs/private-kubernetes/changelog/k8s-1.26.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+---
+
+## Versions
+
+The deployed Kubernetes version varies based on when your cluster is deployed. We try deploying cluster using the lasted patch release of Kubernetes.
+
+Current release is Kubernetes 1.26.6
+
+## Major changes
+  * Support for node autoscaling is added
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/changelog/k8s-1.26.md:14]: Just an opinion, but it is more common and easier to read with the verb first, i.e:
"Added support for node autoscaling
Added this
Removed this
Implemented that"

File: content/en/docs/private-kubernetes/changelog/k8s-1.26.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+More details can be found in [Kubernetes official documentation.](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-29)
+
+  * Flow control `flowcontrol.apiserver.k8s.io/v1beta2`. The replacement `flowcontrol.apiserver.k8s.io/v1beta3` was implemented in Kubernetes 1.26
+
+### Other noteworthy deprecations
+
+Kubernetes beta topology labels on nodes are deprecated and will be removed in a future release. You can follow the list below to see which labels are being replaced:
+
+> **Please note:** The following changes does not have a set Kubernetes release. However, the replacement labels are already implemented.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/changelog/k8s-1.26.md:40]: Maybe change to "The replacement labels are however already implemented". Don't want to start a sentence with "However"

File: content/en/docs/private-kubernetes/getting-started/example-deployment.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
+---
+title: "Your first deployment"
+description: "An example deployment to get started with your Kubernetes cluster"
+weight: 3
+alwaysopen: true
+---
+
+This page will help you getting an deployment up and running and exposed as a load balancer.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/example-deployment.md:8]: *a deployment

File: content/en/docs/private-kubernetes/getting-started/example-deployment.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+> **Note:** This guide is optional and only here to help new Kubernetes users with an example deployment.
+
+Before following this guide you need to have [ordered a cluster](../ordering/) and followed the [Accessing your cluster guide](../accessing/)
+
+You can verify access by running `kubectl get nodes` and if the output is simullar to the example below you are set to go.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/example-deployment.md:14]: *similar

File: content/en/docs/private-kubernetes/getting-started/example-deployment.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+hux-lab1-worker-htfbp          Ready    <none>          15h     v1.27.3
+hux-lab1-worker-k56hn          Ready    <none>          16h     v1.27.3
+```
+
+## Creating an example deployment
+
+To get started we need a deployment to deploy.
+Below we have a deployment called `echoserver` we can use for this example.
+
+1. Start of by creating a file called `deployment.yaml` with the content of the deployment below:
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/example-deployment.md:33]: off *

File: content/en/docs/private-kubernetes/getting-started/example-deployment.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    ❯ kubectl get pod
+    NAME                          READY   STATUS    RESTARTS   AGE
+    echoserver-545465d8dc-4bqqn   1/1     Running   0          51s
+    echoserver-545465d8dc-g5xxr   1/1     Running   0          51s
+    echoserver-545465d8dc-ghrj6   1/1     Running   0          51s
+    ```
+
+## Exposing our deployment
+
+After your pods are created we need to make sure to expose our deployment. In this example we are creating a service of type loadbalancer. If you would have run this application on production you most likely would like to have installed an [ingress controller](../../guides/install-ingress/)
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/example-deployment.md:78]: A bit messy. Suggestion:
"If you run this application in production you would likely install an [ingress controller](../../guides/install-ingress/)

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+---
+title: "Recommendations"
+description: "An list of things we recommend to get the best experience from your Kubernetes cluster"
+weight: 4
+alwaysopen: true
+---
+
+This page describes a list of things that could help you get the best experience out of your cluster.
+
+> **Note:** You do not need to follow this documentation on order to use your cluster
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:10]: *in order

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+## Ingress and cert-manager
+
+To make it easier to expose a lot of applications an ingress controller is a common to deploy.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:14]: Either "is common to deploy" or "is commonly deployed"

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+To get a single IP-address you can point your DNS towards we recommend to deploy an ingress-controller with a service of type Load balancer. [More information regarding load balancers can be found here](#loadbalancers)
+
+In order to automatically generate and update TLS certificates cert-manager is commonly deployed side by side with an ingress controller.
+
+We have created a guide on how to get started with ingress-nginx and Cert-manager that can be found here: [Link to guide](../../guides/ingress/)
+
+## Requests and limits
+
+Below we describe requests and limits briefly. However for a more detailed description or help setting requests and limits we recommend to check out Kubernetes documentation: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:28]: Skip the "However"

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+### Requests
+
+Requests and limits are just to help Kubernetes make better decisions when to schedule and limit your workload.
+
+Requests are important for the scheduler. Requests can be seen as "Amount of resources the pod would utilize during normal operation". This means that the scheduler will allocate the required amount of resources and make sure they are always available to your pod.
+
+If you are using auto scaling requests also help the scheduler when to add new nodes.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:36]: A bit confusing. Is it meant to be "Using auto scaling requests also help"?

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+### Limits
+
+Limits set the maximum allowed resource usage for a pod. This is important to avoid slowdowns in other pods running on the same node.
+
+`CPU` limit you application will be throttled or simply run slower when trying to exceed the limit.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:42]: Maybe change to `CPU` limit. Your application will be...

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+`Memory` limit are another beast. If any pod trying to use more memory than the limit the pod will be Out of memory killed.
+
+## Autoscaling
+
+Autoscaling can be done by scaling nodes and or pods. To get the absolute best experience we recommend a combination of both.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:47]: *and/or

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+## Scaling nodes
+
+We have built in support for the scale nodes up and down. To get started with auto scaling we recommend that you check out our [guide that can be found here](../../guides/autoscaling/)
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:51]: *for scaling nodes up and down

File: content/en/docs/private-kubernetes/getting-started/recommendations.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+Kubernetes documentation have a guide on how to do this [that can be found here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)
+
+In short, node autoscaling is only taken into consideration if you have any pods that cannot be scheduled or if you have set requests on your pods. In order to automatically scale an application pod scaling can make sure you get more pods before reaching your pod limit and if more nodes are needed in order to run the new pods nodes will automatically be added and then later removed when no longer needed.
+
+## Network policies
+
+Network policies can in short be seen as Kubernetes built in firewalls.
+
+Network policy can be used to limit both incoming and outgoing traffic. This allows you to for example specify that only a set of your pods are allowed to talk to the database.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/getting-started/recommendations.md:63]: This allows you for example to*

File: content/en/docs/private-kubernetes/guides/autoscaling.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+Autoscaling currently needs to be configured by Elastx support.
+
+In order to activate auto scaling we need to know what cluster and nodegroup to activate auto scaling on along with the minimum and maximum number of nodes the group can scale to.
+
+Nodes are split into availability zones meaning if you want 3 nodes you get one in each availability zone.
+
+Another example is to have a minimum of 3 nodes and maximum of 7. This would translate to minimum one node per availability zone and maximum 3 in STO1 and 2 in STO2 and STO3 respectively.
+To keep it simple we recommend using increments of 3.
+
+If you are unshare contact out support and we will help you get the configuration you wish for.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/autoscaling.md:44]: *unsure

File: content/en/docs/private-kubernetes/guides/autoscaling.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+
+## How does autoscaling know when to add additional nodes?
+
+Nodes are added once they are needed. and there is 2 scenarious.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/autoscaling.md:48]: Nodes are added once they are needed and there are 2 scenarios:*

File: content/en/docs/private-kubernetes/guides/autoscaling.md [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+1. You have a pod that fails to be scheduled on existing nodes
+2. You schedule pods with requests and the autoscaler senses that your pods won't be able to be scheduled. If a lot of pods are schedule this method can add more than one node at a time.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/autoscaling.md:51]: Second sentence should probably be *scheduled

File: content/en/docs/private-kubernetes/guides/ingress-prep-pre-126.md [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+description: "For clusters created prior to Kubernetes 1.25 some additional configuration is needed to get started with the bundled ingress-nginx controller"
+weight: 5
+alwaysopen: true
+---
+
+> **Note:** This guide is only for clusters created prior to Kubernetes 1.26.
+
+# Preparation and information steps for clusters created prior to Kubernetes 1.26
+
+We run ingress controllers as daemonsets on each of the worker nodes. Providing a
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/ingress-prep-pre-126.md:12]: *This provides
*into your cluster

File: content/en/docs/private-kubernetes/guides/ingress-prep-pre-126.md [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+cheap and easy way to let traffic in to your cluster. To enable traffic to
+reach your worker nodes you need to add rules to the security groups governing
+access to those nodes.
+
+## Security groups
+
+We deliver our clusters unaccessible from the internet, to provide ingress
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/ingress-prep-pre-126.md:19]: *inaccessible

*from the internet. To provide

File: content/en/docs/private-kubernetes/guides/ingress-prep-pre-126.md [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+access you need to define rules allowing such traffic.
+
+To do so, log in to [Elastx OpenStack IaaS](https://ops.elastx.cloud/). Once logged
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/ingress-prep-pre-126.md:22]: *into

File: content/en/docs/private-kubernetes/guides/ingress.md [hunk 23]
```diff
@@ -5,57 +5,23 @@ weight: 5
...
-    Rule: Custom TCP Rule
-    Direction: Ingress
-    Open Port: Port
-    Port: 443
-    Remote: CIDR
-    CIDR: 0.0.0.0/0
+Prior to Kubernetes 1.26 Elastx Kubernetes CaaS clusters was shipped with [NGINX Ingress Controller](https://kubernetes.github.io/ingress-nginx/)
+and [cert-manager](https://cert-manager.io/docs/).
+
+For cluster running Kubernetes 1.26 or later we will publish a guide on how to install and upgrade your ingress and cert-manager deployments
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/ingress.md:11]: Maybe add "... how to install and upgrade your ingress and cert-manager deployments yourselves" at the end to clarify this is something the customer will have to do moving forward.

File: content/en/docs/private-kubernetes/guides/loadbalancer.md [hunk 24]
```diff
@@ -207,3 +207,9 @@ zones. This is a limitation imposed by the OpenStack Octavia project.
 Reconfiguring the load balancers using annotations is not as dynamic and smooth
 as one would hope. For now, to change the configuration of a load balancer the
 service needs to be deleted and a new one created.
+
+## Loadbalancer protocols
+
+Loadbalancers have support for multiple protocols. In general we would recommend everyone to try avoiding http and https due to a few limitations and simply they does not perfom as well as other protocols.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/guides/loadbalancer.md:213]: *simply because they do not perform

File: content/en/docs/private-kubernetes/guides/ingress.md (hunk 23 above)
Comment (SereneVirus) [reply content/en/docs/private-kubernetes/guides/ingress.md:11]: This was also covered in content/en/docs/private-kubernetes/overview.md
Maybe point the documentation to the PCaaS overview instead of having almost the same information written twice.

File: content/en/docs/private-kubernetes/overview.md [hunk 25]
```diff
@@ -84,3 +75,13 @@ rebuilding the cluster.
...
+
+Since Elastx Private Kubernetes 2.0 we offer auto scaling of workload nodes. This is based on resource requests meaning it relies on setting realistc requests on your workload. All configuration is currently a manual process involving contact with either our sales department or our support.
+
+### Optional features
+
+Prior to Kubernetes relase 1.26 we offered NGINX Ingress Controller and cert-manager as optional features.
+
+If you have an Ingress or certmanager installed by Elastx prior to Kubernetes 1.26 you need to manually upgrade them starting from the Kubernetes 1.26 release. In order to do this you can follow the [instruction for ingress here:](../guides/install-ingress/) and [instructions for certmanager here](../guides/install-certmanager/)
+
+If you are ordering a new Kubernetes cluster you need to install an ingress and certmanager on your own behalf. We have however created a guide on how to install Nginx Ingress Controller and cert-manager. We will also maintain upgrade instructions comming out with new relases over time. If you instead wish to use your own ingress contoller you can simply ignore this part.
```
Comment (SereneVirus) [inline content/en/docs/private-kubernetes/overview.md:87]: *coming

**Review by SereneVirus — CHANGES_REQUESTED** [review]
Since this is customer facing I have made many grammatical suggestions and corrected some errors throughout comments.

File: content/en/docs/private-kubernetes/guides/ingress.md (hunk 23 above)
Comment (huxcrux) [reply content/en/docs/private-kubernetes/guides/ingress.md:11]: I think it could be a good thing to have both here and the ovewrview page for now. The overview page will most likely get cleaned up as soon as all clusters are upgraded to 1.26 or later

File: content/en/docs/private-kubernetes/getting-started/recommendations.md (hunk 12 above)
Comment (huxcrux) [reply content/en/docs/private-kubernetes/getting-started/recommendations.md:36]: Reprashed this sentence, is ot more clear now?

**Review by SereneVirus — APPROVED** [review]

**Review by SereneVirus — APPROVED** [review]
Looking good now!

File: content/en/docs/private-kubernetes/guides/autoscaling.md [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+hux-lab1-worker-k56hn          Ready    <none>          16h     v1.27.3
+hux-lab1-nodegroup2-33hbp      Ready    <none>          15h     v1.27.3
+hux-lab1-nodegroup2-54j5k      Ready    <none>          16h     v1.27.3
+```
+
+## How to activate autoscaling?
+
+Autoscaling currently needs to be configured by Elastx support.
+
+In order to activate auto scaling we need to know `clustername` and `nodegroup` with two values for `minimum/maximum` number of desired nodes. Currently we cannot have minimum set below 3 nodes however this is subnject to vhange in the future.
```
Comment (zrk02) [inline content/en/docs/private-kubernetes/guides/autoscaling.md:41]:
```suggestion
In order to activate auto scaling we need to know `clustername` and `nodegroup` with two values for `minimum/maximum` number of desired nodes. Currently we have a minimum set to 3 nodes however this is subject to change in the future.
```
